src/invidious/helpers/helpers.cr

```diff
@@ -64,11 +64,14 @@ end
 class Config
   include YAML::Serializable
 
-  property channel_threads : Int32 = 1             # Number of threads to use for crawling videos from channels (for updating subscriptions)
-  property feed_threads : Int32 = 1                # Number of threads to use for updating feeds
-  property output : String = "STDOUT"              # Log file path or STDOUT
-  property log_level : LogLevel = LogLevel::Info   # Default log level, valid YAML values are ints and strings, see src/invidious/helpers/logger.cr
-  property db : DBConfig                           # Database configuration
+  property channel_threads : Int32 = 1           # Number of threads to use for crawling videos from channels (for updating subscriptions)
+  property feed_threads : Int32 = 1              # Number of threads to use for updating feeds
+  property output : String = "STDOUT"            # Log file path or STDOUT
+  property log_level : LogLevel = LogLevel::Info # Default log level, valid YAML values are ints and strings, see src/invidious/helpers/logger.cr
+  property db : DBConfig? = nil                  # Database configuration with separate parameters (username, hostname, etc)
+
+  @[YAML::Field(converter: Preferences::URIConverter)]
+  property database_url : URI? = nil               # Database configuration using 12-Factor "Database URL" syntax
```

Comment:
Made these nullable since we now allow either one of them to be provided (with the `database_url` taking precedence).